features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -658,7 +660,10 @@ nsapi_error_t mbed_lwip_bringup_2(bool dhcp, bool ppp, const char *ip, const cha
     if (!netif_is_link_up(&lwip_netif)) {
         if (sys_arch_sem_wait(&lwip_netif_linked, 15000) == SYS_ARCH_TIMEOUT) {
             if (ppp) {
-                ppp_lwip_disconnect();
+                err_t err = ppp_lwip_disconnect();
```

Comment:
Not sure about this and the similar change below - I think we always want to return NSAPI_ERROR_NO_CONNECTION, not whatever oddities PPP might throw up during attempted shutdown.